### DISCUSSION TITLE
Email alert

### DISCUSSION THREAD

**Discussion by nereauweb** [abraunegg/onedrive#2122] [General]
Would it be possible to somehow have linux onedrive service (monitor mode) send an email alert (or an hookable system alert) when for any reason (especially expired token) synchronization stops? 
I use it on a centos server to sync the samba shared folder of an organization, to assure backup and operational continuity if the server fails, but I noticed it stops, making it useless if not regularly (and manually) checked...

**Comment by abraunegg** [comment]
@nereauweb 
> but I noticed it stops, making it useless if not regularly .. checked

Firstly .. what do the application logs say as to why it stopped? If the logs do not conain much, increase logging verbosity and fix that problem first. 

> Would it be possible to somehow have linux onedrive service (monitor mode) send an email alert (or an hookable system alert) when for any reason (especially expired token) synchronization stops?

Your best avenue is to enable logging to a log file, then, using crontab or similar, search the logfile for events to trigger generating and email alert.

If you are using systemd, you can reconfigure the systemd unit to send emails on failure - see https://serverfault.com/questions/694818/get-notification-when-systemd-monitored-service-enters-failed-state for ideas.

Adding into the application to send an email will be problematic for a number of reasons - smtp, new authentication challenges, resolution issues etc .. and there are much better ways to handle this as per above.

**Reply by nereauweb** [reply]
Thanks, I will look into that approach
